### ISSUE TITLE
Retrieve Original Filenames of OleObject.bin files

### ISSUE THREAD

**Issue by SmithCJ47** [decalage2/oletools#726]
**Is your feature request related to a problem? Please describe.**
I have a word document that has 100+ pdfs attached to it. I would like to pull all the pdfs from the document and archive them in a folder hierarchy for easier browsing. I have successfully used the olefile package to pull all the pdfs and save them to a particular folder. My problem is that olefile package does not pull out the embedded filename. Current code saves it as documentXXX.pdf. 

**Describe the solution you'd like**
I would like to be able to retrieve the original filename of the pdf when stored in docx file. Then use that string to name the retrieved pdf file.

**Describe alternatives you've considered**
I've tried opening up a olefile.OleFileIO() object to retrieve the metadata but have been unable to find the filename. Alternatively, I've searched through the xml code to see if the filename is embedded there somewhere. (It must be, the name is displayed on the word document icon that will open up the embedded pdf). But I have been unsuccessful in determining the location of the stored filename. The closest thing I found was in word/_rels/document.xml.rels which linked the ID of the embedded object to the OldObjectXX.bin file.

**Additional context**
Thank you for making these tools. I have found them to be very helpful! I'm also willing to help contribute where I can to implement this feature request.

**Comment by decalage2** [comment]
Hi @SmithCJ47, did you try oleobj by any chance?

**Comment by SmithCJ47** [comment]
Thank you for the suggestion. I tried to use oleobj with no luck. Still not able to extract original pdf filenames from pdfs encoded as oleobjects. I tried using the Olebrowse to open the oleobject, to which it gave hex data. But the filename was not included there either. When I use olemeta, I get this: 
![image](https://user-images.githubusercontent.com/67006779/150581743-f96035b9-edaf-4f04-87c9-a630c9047fb7.png)

I put together a test file to try out various solutions. I'll include it here so you don't have to reproduce the problem on your own.
[Test.docx](https://github.com/decalage2/oletools/files/7915890/Test.docx)

**Comment by decalage2** [comment]
I checked your test file, and I think the PDF filename is stored nowhere in the OLE objects nor in the XML files of the document. I searched both ascii and unicode strings. 
The only place where the PDF filename (with "Anaconda" in it) appears, is in the embedded EMF file that contains the icon used to display the OLE object in Word: word\media\image1.emf and image2.emf.
![image](https://user-images.githubusercontent.com/5989656/150657563-0d3977de-4364-44d5-879d-2dafab23cc4f.png)

But it's just a string displayed with the image, not a filename that you can easily extract in general.
Here it is in a hex viewer:
![image](https://user-images.githubusercontent.com/5989656/150657599-ecc9c313-2995-42f9-a8f7-d478440019f5.png)

**Comment by decalage2** [comment]
I propose to close this issue, unless you find another place where the filename is stored, and which could be parsed properly.

**Comment by SmithCJ47** [comment]
I agree, let's close the issue. I've searched and have been unable to find the filename anywhere else. Thank you so much for your time on looking into this. I really appreciate your help! -Also, thank you for this great package, super useful!
